engine/src/main/java/com/sandboni/core/engine/Processor.java

```diff
@@ -185,15 +185,26 @@ private Builder getBuilder(Context context) {
     }
 
     private boolean moduleContainsChanges(ChangeScope<Change> changeScope) {
+        log.debug("Check if moduleContainsChanges");
         return scopeFilter.isInScope(changeScope,
             Stream.of(arguments.getSrcLocation())
                 .map(File::new)
                 .collect(Collectors.toSet()));
     }
 
     private void executeFinders(Context context) {
-        FinderExecutor finderExecutor = new FinderExecutor(context);
-        finderExecutor.execute(finders);
+        log.debug("executeFinders starts");
+        finders.parallelStream().forEach(f -> {
+            long start = System.nanoTime();
+            log.debug("[{}] Finder {} started", Thread.currentThread().getName(), f.getClass().getSimpleName());
+
+            Context localContext = context.getLocalContext();
+            f.findSafe(localContext);
+            context.addLinks(localContext.getLinks().toArray(Link[]::new));
+
+            log.debug("[{}] Finder {} finished in {} milliseconds", Thread.currentThread().getName(), f.getClass().getSimpleName(), elapsedTime(start));
+        });
+        log.debug("executeFinders ends");
```

Comment:
can we test this on jules before it goes live?

Reply:
Not before going live. We can't use snapshots versions in Jules.